earthmover/earthmover.py

```diff
@@ -121,11 +122,23 @@ def load_project_configs(self, filepath: str):
         """
         configs = JinjaEnvironmentYamlLoader.load_project_configs(filepath, params=self.params)
 
-        # Update project parameter defaults from the template, if any
+        # 1. Update project parameter defaults from the template, if any
         for key, val in configs.get("parameter_defaults", {}).items():
             self.params.setdefault(key, val)
 
-        # Prepend package macros to the project macro string. Later macro definitions in the string will overwrite earlier ones
+        # 2. There may be config keys that expect an environment variable but for which a default is also defined.
+        #     If no env var was passed by the user, apply the default value.
+        env_var_pattern = r'^\$\{[A-Z_]+\}$'
+        regex = re.compile(env_var_pattern)
+
+        for key, val in configs.items():
```

Comment:
Not a problem right now, but if in the future we add any nested [`configs`](https://github.com/edanalytics/earthmover#config), this would only apply to the top level... we may want to add recursion later.

Example:
```yaml
config:
  some_new:
    nested_config: ${MY_VAR}
  parameter_defaults:
    MY_VAR: 'my_value'
```

Reply:
It's a fair point but for simplicity I'll ignore this case for now